tests/integration/testlib/examplelib.py

```diff
@@ -41,7 +41,7 @@ def example_state(initial, cleanup=None):
         yield desired_state
     finally:
         if cleanup:
-            libnmstate.apply(load_example(cleanup))
+            libnmstate.apply(load_example(cleanup), verify_change=False)
```

Comment:
IMHO it is wrong for the examples not to verify that the cleanup example works as well. It documents to our users how interfaces can be removed and therefore it should be expected to always work. Why do you want to disable it?

Reply:
The test is responsible to assert if the action succeeded or not.

The action of cleanup should always be done, even if the end result is not as expected because we *must* cleanup at the test end. If there is a bug at the creation, we want the cleanup to happen and do its best.
So I prefer something over nothing.

Reply:
So how about

```
try:
    libnmstate.apply(load_example(cleanup))
except:
    libnmstate.apply(load_example(cleanup), verify_change=False)
    raise
```

This way the test should fail if the cleanup works (it does , doesn't it?) and it will still try to cleanup with libnmstate a little bit more. It would probably be better to do a proper cleanup without using libnmstate by just removing all profiles/interfaces, see https://lists.fedorahosted.org/archives/list/nmstate-devel@lists.fedorahosted.org/thread/ZHTJTFZZR6YV3FLMX2AUYQA75GH5SSDZ/

Reply:
As I mentioned before, the cleanup should occur in the best way possible and if it did not do the job well, it should fail on an assertion.
Your suggestion may work, it is a bit wasteful IMO, but if it works, I do not mind.

Regarding cleaning up NM profiles directly... it is possible, but it sounds to me a cleanup of a cleanup. If we do it right in a test, there should be no leftovers.
I also prefer not to depend on NM being the backend, at least the tests should now depend on that.. as much as we possible can (there are always exceptions).

Reply:
IMHO, @tyll 's proposed solution looks good to me. I think it is good to know that we have known issues on the cleanup verification. About the profile, I think @EdDev is right.

Reply:
> Regarding cleaning up NM profiles directly... it is possible, but it sounds to me a cleanup of a cleanup. If we do it right in a test, there should be no leftovers.

Please read https://lists.fedorahosted.org/archives/list/nmstate-devel@lists.fedorahosted.org/thread/ZHTJTFZZR6YV3FLMX2AUYQA75GH5SSDZ/ - there is a problem with using checkpoints during cleanup therefore it would not be a second cleanup/teardown but the only one. However, for the examples we also want to test that the examples to remove something also work and probably in other tests, too. But these tests should probably still have a teardown cleanup phase.

> I also prefer not to depend on NM being the backend, at least the tests should now depend on that.. as much as we possible can (there are always exceptions).

Once we have another backend, it could be added to the cleanup tool so it does the appropriate cleanup depending on the tested backend.